File: client-flutter/lib/models/Exercise.dart
class Exercise {
  final int id;
  final String name;
  final String targetArea;
  final String difficulty;
  final String description;
  final String videoUrl;
  final int duration;
  Exercise({
    required this.id,
    required this.name,
    required this.targetArea,
    required this.difficulty,
    required this.description,
    required this.videoUrl,
    required this.duration,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'],
      name: json['name'],
      targetArea: json['targetArea'],
      difficulty: json['difficulty'],
      description: json['description'],
      videoUrl: json['videoUrl'],
      duration: json['duration'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
    };
  }
}

File: server/src/main/java/org/server/workout/entities/Exercise.java
package org.server.workout.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Exercise {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id", nullable = false)
    private int id;
    @Column(name = "name", nullable = false, length = 100)
    private String name;
    @Column(name = "target_area", nullable = true, length = 100)
    private String targetArea;
    @Column(name = "difficulty", nullable = true, length = 50)
    private String difficulty;
    @Column(name = "description", nullable = true, length = -1)
    private String description;
    @Column(name = "video_url", nullable = true, length = 255)
    private String videoUrl;

}
